modules/internal/LocaleModelHelpSetup.chpl

```diff
@@ -73,6 +73,10 @@ module LocaleModelHelpSetup {
     forall locIdx in dst.chpl_initOnLocales() with (ref root_accum) {
       const node = new locale(new unmanaged LocaleModel(new locale(dst)));
       dst.myLocales[locIdx] = node;
+      // set the defaultLocale here to avoid extra ons during startup
+      if locIdx == 0 {
+        chpl_set_defaultLocale(node);
```

Comment:
May be missing something obvious, but if the defaultLocale is now localeprivate, doesn't it need to be set on all locales?

Reply:
Nope, you are absolutely right. The last commit makes some changes to make this "setting default locale" logic to be more similar to "setting root locale" and updates the test to actually check the defaultLocale value. I think my gasnet testing must have failed because of that. I'll rerun another gasnet paratest